### ISSUE TITLE
Running with an AMD CPU

### ISSUE THREAD

**Issue by Kircheneer** [ipspace/netsim-tools#9]
I had to configure the following in `netsim/templates/provider/libvirt/vsrx-domain.j2`, otherwise the vSRX got stuck on
```Booting [/packages/sets/pending/boot/os-kernel/kernel]...```

This sets the `--cpu` parameter for `virt-install` to `qemu64` instead of the default `host`.

```
{{ name }}.vm.provider :libvirt do |domain|
    domain.default_prefix = "#{cwd}_"
    domain.cpus = 2
    domain.memory = 4096
    domain.disk_bus = "ide"
 +  domain.cpu_mode = "custom"
end
```

I was wondering if you have any experiences with AMD/intel differences in this regard. I could submit a PR that insert this line whenever a AMD CPU was used. The only problem is that I have no way to test whether this happens for all/most/any AMD setups except for mine because I have no other system available.

**Comment by ipspace** [comment]
Let me see if the same setting works on my Intel NUC. If it does, we can make it default and move on. It's not like we're aiming for a perfect performance-optimized solution ;)

**Comment by ipspace** [comment]
This is what happens on Intel NUC with default install of various packages:

```
There was an error talking to Libvirt. The error message is shown
below:

Call to virDomainCreateWithFlags failed: the CPU is incompatible with host CPU: Host CPU does not provide required features: svm
```

So yes, we should only do that on an AMD CPU, and I can put a mention in platform support document.

I created **dev_0.6.2** branch to include your fix and a few fixes I added for simpler FRR deployment (I hate committing small fixes directly into the master branch). Please submit a PR for that branch. Thank you!

**Comment by Kircheneer** [comment]
One small issue I see with that. The performance is absolutely abismal, boot takes several minutes, ssh login takes 3 seconds and show version takes 5 seconds. My knowledge of CPU architectures and the like is unfortunately quite little so I don't really know where to go from here. We could include the fix and put a note in the platform support document stating that performance may be bad for vSRX under AMD CPUs?

**Comment by ipspace** [comment]
In that case, let's not try to fix it. I'll add a note to the platforms document, explain your fix, and ask anyone with more experience with libvirt to fix it ; Will also try to ask my Juniper contacts whether they could get me in touch with someone who knows more.

**Comment by Kircheneer** [comment]
I'm having the exact same problem for NXOS. Shall I submit a similar patch for that as well?

**Comment by ipspace** [comment]
Sure. Thank you!

**Comment by Kircheneer** [comment]
Fixed in 0.6.2.
